### PR TITLE
added keras and tensorflow version updates

### PR DESCRIPTION
**Pull request by prprakash02**
_The pull request contains changes to the following files:_

lib_data.py:

- imgaug has been replaced with RandAugment from keras_cv, which relies on tensorflow v2.16.1
- An additional check for the magnitude parameter in RandAugment is added, warning is raised if out of bounds (warnings will be generated if ignore is modified in future iterations)
- sklearn.preprocessing dependencies are now up-to date

lib_model.py:

- tensorflow_addons is now replaced with keras_cv and there are two possible loss functions (added in comments). The implemented FocalLoss() is the same as the focal loss used in tensorflow_addons (please click the [link](https://keras.io/api/keras_cv/losses/focal_loss/#:~:text=References-,Focal%20Loss%20paper,-Standalone%20usage%3A) for more info)

Kind Regards,
Prakash

### CODE REVIEW

**Comment by zaandahl** [comment]
Updated base mewc-flow package to include keras-cv. PR passes workflow, approving merge
